commerce/domain/shipping/splitshipment/SplitShipmentUnitTest.cls

```diff
@@ -0,0 +1,244 @@
+        insert(testProduct1);
+
+        CartItem cartItem1 = new CartItem(
+                Name = CART_ITEM1_NAME,
+                CartId = cartId,
+                CartDeliveryGroupId = deliveryGroupId,
+                Quantity = 10.0,
+                SKU = SKU1_NAME,
+                Product2Id=testProduct1.Id,
+                Type = CartExtension.SalesItemTypeEnum.PRODUCT.name());
+        insert cartItem1;
+
+        // Configure the second product as conveyable
+        Product2 testProduct2 = new Product2(name='My Product 2', IsConveyable__c = true);
+        insert(testProduct2);
+
+        CartItem cartItem2 = new CartItem(
+                Name = CART_ITEM2_NAME,
+                CartId = cartId,
+                CartDeliveryGroupId = deliveryGroupId,
+                Quantity = 2.0,
+                SKU = SKU2_NAME,
+                Product2Id=testProduct2.Id,
+                Type = CartExtension.SalesItemTypeEnum.PRODUCT.name());
+        insert cartItem2;
+
+        // Configure the third product as non-conveyable
+        Product2 testProduct3 = new Product2(name='My Product 3', IsConveyable__c = false);
+        insert(testProduct3);
+
+        CartItem cartItem3 = new CartItem(
+                Name = CART_ITEM3_NAME,
+                CartId = cartId,
+                CartDeliveryGroupId = deliveryGroupId,
+                Quantity = 5.0,
+                SKU = SKU3_NAME,
+                Product2Id=testProduct3.Id,
+                Type = CartExtension.SalesItemTypeEnum.PRODUCT.name());
+        insert cartItem3;
+
+        // Return a map where the key -> value is: CartItemId -> ProductId
+        return new Map<ID, ID>{ cartItem1.Id => testProduct1.Id, cartItem2.Id => testProduct2.Id, cartItem3.Id => testProduct3.Id };
+    }
+
+    /**
+     * @description Create an ItemArrange object, provided the cartItemId and quantity
+     * @param cartItemId ID of the CartItem
+     * @param deliveryGroupId ID of the CartDeliveryGroup
+     * @param productId ID of the Product2
+     *
+     * @return CartExtension.ItemArrange Holds details of how to arrange an item between CartDeliveryGroups
+     */
+    private static CartExtension.ItemArrange createItemArrange(ID cartItemId, ID deliveryGroupId, ID productId) {
+        CartExtension.ItemArrange.Builder builder = new CartExtension.ItemArrange.Builder()
+            .withCartItemId(cartItemId)
+            .withProductId(productId);
+
+        if (deliveryGroupId != null) {
+            builder.withDeliveryGroupId(deliveryGroupId);
+        } else {
+            builder
+                .withDeliverToCity('Foxborough')
+                .withDeliverToCountry('United States')
+                .withDeliverFromFirstName('Tom')
+                .withDeliverFromLastName('Brady')
+                .withDeliverToName('Rob Gronkowski')
+                .withDeliverToPostalCode('02035')
+                .withDeliverToState('Massachusetts')
```

Comment:
The address should be really Raymond James Stadium, Tampa, FL, 33602 🤣  Just kidding